dashboard/src/components/home/CreateWorkspaceForm/CreateWorkspaceForm.tsx

```diff
@@ -0,0 +1,170 @@
+import Form from '@/components/common/Form';
+import Button from '@/ui/v2/Button';
+import Input from '@/ui/v2/Input';
+import { useInsertWorkspaceMutation } from '@/utils/__generated__/graphql';
+import { slugifyString } from '@/utils/helpers';
+import { nhost } from '@/utils/nhost';
+import { toastStyleProps } from '@/utils/settings/settingsConstants';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useRouter } from 'next/router';
+import { FormProvider, useForm } from 'react-hook-form';
+import { toast } from 'react-hot-toast';
+import * as Yup from 'yup';
+
+export interface CreateWorkspaceFormProps {
+  /**
+   * Determines whether the form is disabled.
+   */
+  disabled?: boolean;
+  /**
+   * Submit button text.
+   *
+   * @default 'Save'
+   */
+  submitButtonText?: string;
+  /**
+   * Function to be called when the form is submitted.
+   */
+  onSubmit?: () => void;
+  /**
+   * Function to be called when the operation is cancelled.
+   */
+  onCancel?: VoidFunction;
+}
+
+export interface CreateWorkspaceFormValues {
+  /**
+   * New workspace name.
+   */
+  newWorkspaceName: string;
+}
+
+const validationSchema = Yup.object().shape({
+  newWorkspaceName: Yup.string()
+    .required('Workspace name is required.')
+    .min(4, 'The new Workspace name must be at least 4 characters.')
+    .max(32, "The new Workspace name can't be longer than 32 characters.")
+    .test('canBeSlugified', 'This is not a valid JSON.', (value) => {
+      try {
+        const slug = slugifyString(value);
+        if (slug.length < 4 || slug.length > 32) {
+          throw new Error(
+            "This field should be at least 4 characters and can't be longer than 32 characters.",
+          );
+        }
+        return true;
+      } catch (error) {
+        return false;
+      }
+    }),
+});
+
+export default function CreateWorkspaceForm({
+  disabled,
+  onSubmit,
+  onCancel,
+  submitButtonText = 'Save',
+}: CreateWorkspaceFormProps) {
+  const [insertWorkspace, { client }] = useInsertWorkspaceMutation();
+  const router = useRouter();
+
+  const form = useForm<CreateWorkspaceFormValues>({
+    defaultValues: {
+      newWorkspaceName: '',
+    },
+    resolver: yupResolver(validationSchema),
+  });
+
+  const {
+    register,
+    formState: { dirtyFields, isSubmitting, errors },
+  } = form;
+  const isDirty = Object.keys(dirtyFields).length > 0;
+
+  const currentUser = nhost.auth.getUser();
```

Comment:
You can make use of our `useUserData()` hook.

Reply:
Just changed it, but as a note for us: 

`useUserData()` returns `any` while `nhost.auth.getUser();` returns a `User`

Reply:
Hmm, the type seems to be lost somewhere along the way, because in `packages/react/src/useUserData.ts` I see that the function returns a `User`. 🤔 

Do you have any ideas @plmercereau?

This is what I've found: `export declare const useUserData: () => import("@nhost/hasura-auth-js").User | null;`

Maybe we need to explicitly cast the return value to `User` to import it from `@nhost/nhost-js` instead of `@nhost/hasura-auth-js` which is not a dependency of `@nhost/react`?

Reply:
Created an issue out of the comments: #1484